math/mode.go

```diff
@@ -0,0 +1,50 @@
+// mode.go
+// author(s): [CalvinNJK] (https://github.com/CalvinNJK)
+// description: Finding Mode Value In an Array
+// see mode.go
+
+package math
+
+import (
+	"errors"
+
+	"github.com/TheAlgorithms/Go/constraints"
+)
+
+// ErrEmptySlice is the error returned by functions in math package when
+// an empty slice is provided to it as argument when the function expects
+// a non-empty slice.
+var ErrEmptySlice = errors.New("empty slice provided")
+
+func Mode[T constraints.Number](numbers []T) (T, error) {
+
+	n := len(numbers)
+
+	maxCount := 0
+	var maxValue T = 0
+
+	if n == 0 {
+		return 0, ErrEmptySlice
+	} else {
```

Comment:
This if block should go before definition of `maxCount` and `maxValue` (line 22) since everything you need for this block is already present at that point.

After that you don't need the else statement as mentioned in the other comment.